src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -112,6 +127,56 @@ class RegularSync(
       }
   }
 
+  /**
+    * Handles NewHashesMessege, should only cover this message when we are top of the chain
+    */
+  def handleNewBlockHashesMessages: Receive = {
+    case MessageFromPeer(NewBlockHashes(hashes), peerId) =>
+      val maybePeer = peersToDownloadFrom.find(peer => peer._1.id == peerId)
+      //we allow asking for new hashes when we are not syncing and we can download from specified peer ,we are
+      //top of the chain and not resolving branches currently
+      if (notDownloading() && topOfTheChain && maybePeer.isDefined && !resolvingBranches) {
```

Comment:
Do we maybe need to add this check also for handling the NewBlock messages? Maybe this check could then be added to the notDownloading() function?

Reply:
actually, I think we don't need it in terms of safety ( newBlock don't start any download or schedule resume), but it may be good idea to just move this condition to `notDownloading()`